Lib/test/test_sax.py

```diff
@@ -1307,6 +1306,23 @@ def test_expat_locator_withinfo_nonascii(self):
         self.assertEqual(parser.getSystemId(), fname)
         self.assertEqual(parser.getPublicId(), None)
 
+    def test_qualified_names(self):
+
+        class Handler(ContentHandler):
+            def startElementNS(self, name, qname, attrs):
+                self.qname = qname
+
+        for xml_s, expected_qname in zip(["<Q:E xmlns:Q='http://example.org/testuri'/>", "<E xmlns='http://example.org/testuri'/>", "<E />"], ["Q:E", "E", "E"]):
```

Comment:
I would consider this variant much more readable:

```suggestion
        for xml_s, expected_qname in (
            ("<Q:E xmlns:Q='http://example.org/testuri'/>", "Q:E"),
            ("<E xmlns='http://example.org/testuri'/>", "E"),
            ("<E />", "E"),
        ):
```